### ISSUE TITLE
filetypes should become a plugin

### ISSUE THREAD

**Issue by Akuli** [Akuli/porcupine#134]
similarly to `plugins/keybindings.py`

**Comment by Akuli** [comment]
`-n` option on command line needs filetypes, so implementing this means that plugins must be able to add custom command line options (e.g. access the `argparse.ArgumentParser`)

that means that plugin loading has to be done in two steps:
1. importing plugins and running something that happens before argument parsing and hence before tkinter widgets get created (could be called `setup_arguments()` or something)
2. running the `setup()` functions

tl;dr pluginloader would get more complicated

alternatively, argument parsing stuff could accept `-n <string>` and just store the strings somewhere, but then filetypes stuff would be weirdly split into two places, some stuff in a plugin and other stuff not in plugins
